### PR TITLE
Fibonacci problem

### PR DESCRIPTION
**Pull request by avenmia**
Please include a summary of the change. Please also include relevant motivation and context (if applicable).

- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the comments
- [ ] I have made corresponding changes to the README.md

### CODE REVIEW

File: Algorithms/Problems/Fibonacci/Fibonacci.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Text;
+
+namespace Algorithms.Problems.Fibonacci
+{
+    public class Fibonacci
```
Comment (siriak) [inline Algorithms/Problems/Fibonacci/Fibonacci.cs:8]: Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6395663)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- Algorithms/Problems/Fibonacci/Fibonacci.cs  3
- Algorithms/Problems/Fibonacci/FibonacciMemo.cs  4
- Algorithms.Tests/Problems/FibonacciTests/FibonacciMemoTests.cs  1
- Algorithms.Tests/Problems/FibonacciTests/GetFibonacciNumberTests.cs  1
         

Clones added
============
- Algorithms.Tests/Problems/FibonacciTests/FibonacciMemoTests.cs  1
- Algorithms.Tests/Problems/FibonacciTests/GetFibonacciNumberTests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6395663&bid=20739990)

File: Algorithms/Problems/Fibonacci/FibonacciMemo.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Text;
+
+namespace Algorithms.Problems.Fibonacci
+{
+    public class FibonacciMemo
```
Comment (siriak) [inline Algorithms/Problems/Fibonacci/FibonacciMemo.cs:9]: Codacy found an issue: [Add a 'protected' constructor or the 'static' keyword to the class declaration.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6395663)

**Comment by siriak** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Algorithms/Problems/Fibonacci/Fibonacci.cs  3
- Algorithms/Problems/Fibonacci/FibonacciMemo.cs  4
- Algorithms.Tests/Problems/FibonacciTests/FibonacciMemoTests.cs  1
- Algorithms.Tests/Problems/FibonacciTests/GetFibonacciNumberTests.cs  1
         

Clones added
============
- Algorithms.Tests/Problems/FibonacciTests/FibonacciMemoTests.cs  1
- Algorithms.Tests/Problems/FibonacciTests/GetFibonacciNumberTests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6395663&bid=20739990)

**Comment by siriak** [comment]
Hi! Fibonacci sequence is already present in [Algorithms/Sequences/FibonacciSequence](https://github.com/TheAlgorithms/C-Sharp/blob/master/Algorithms/Sequences/FibonacciSequence.cs)

**Comment by avenmia** [comment]
Hi @siriak ! I saw the sequence is present, but are the solutions to getting the Nth fibonacci number there and I missed it? This problem was taught to me as a good example of how memoization can significantly reduce runtime.

**Comment by siriak** [comment]
Memoization can be very useful indeed. What you have implemented is called tabulation though. This is a bottom-up approach as opposed to memoization which is top-down. See [here](https://programming.guide/dynamic-programming-vs-memoization-vs-tabulation.html). I see that having examples of memoization and tabulation would be good, I just don't think that Fibonacci numbers are a good problem for that... I mean that while applying it to Fibonacci numbers demonstrates techniques in question, it's completely impractical because it requires O(n) space while there is already an implementation with O(1) space, which is way shorter, simpler, and faster. It is just superior in all ways. It'd be good if you could come up with a problem where memoization is a good solution.
By the way, if you need to get the n-th Fibonacci number, just use Linq:
`fibonacciSequence.Skip(n).First()`;

**Comment by avenmia** [comment]
I learned a lot, thank you!

**Pull request closed without merging**